community/kernel/src/main/java/org/neo4j/kernel/impl/index/schema/IndexPartsCache.java

```diff
@@ -0,0 +1,54 @@
+/*
+ * Copyright (c) 2002-2018 "Neo Technology,"
+ * Network Engine for Objects in Lund AB [http://neotechnology.com]
+ *
+ * This file is part of Neo4j.
+ *
+ * Neo4j is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+package org.neo4j.kernel.impl.index.schema;
+
+import java.util.Iterator;
+import java.util.concurrent.ConcurrentHashMap;
+import java.util.concurrent.locks.Lock;
+import java.util.concurrent.locks.ReentrantLock;
+
+abstract class IndexPartsCache<KEY,T> implements Iterable<T>
```

Comment:
I'm not sure I like this breakout. For two reasons:
1. I think it's outside the scope of this PR.
2. It feels like there are much more that should be done to clean this part of the code up and to me this "half way" change makes it even harder to understand because of the direct access of lock and cache in subclasses.|

My suggestion is that we do a proper refactoring after GA.

Reply:
Did it because there was one name change related to a previous commit in this PR that was previously only done in one of the two duplicated classes, but forgotten the another. The duplication is silly and I don't think we're off worse with this change. If you really think we're in a worse position now then I can revert it.